source/common/ssl/ssl_socket.cc

```diff
@@ -17,6 +17,35 @@ using Envoy::Network::PostIoAction;
 namespace Envoy {
 namespace Ssl {
 
+namespace {
+// This SslSocket will be used when SSL secret is not fetched from SDS server.
+class NotReadySslSocket : public Network::TransportSocket, public Connection {
```

Comment:
I don't think you need this to be a Ssl::Connection. just inherit TransportSocket is enough.

Reply:
Done. Thanks.